src/node_api_jsrt.cc

```diff
@@ -241,23 +241,49 @@ class ExternalCallback {
   void* _data;
 };
 
+inline napi_status JsCopyUtf8Bytes(JsValueRef key,
+                                   char** buffer, size_t* copiedBytesCount) {
+  int strLength = 0;
+  CHECK_JSRT(JsGetStringLength(key, &strLength), napi_string_expected);
+
+  // assume string contains ascii characters only
+  *buffer = reinterpret_cast<char*>(malloc(strLength + 1));
```

Comment:
CHAKRA_VERIFY is a fast fail, CHECK_JSRT returns the error code. Memory leaks are less of an issue when the process is dying anyway.